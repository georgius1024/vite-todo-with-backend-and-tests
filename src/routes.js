import Todos from "./views/Todos.vue";
import Todo from "./views/Todo.vue";
import TodoNew from "./views/TodoNew.vue";

export default [
  { path: "/new", component: TodoNew },
  { path: "/:id", component: Todo },
  { path: "/", component: Todos },
];
