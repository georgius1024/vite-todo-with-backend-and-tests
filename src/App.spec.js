import { describe, it, expect, vi, beforeEach } from "vitest";
import { mount } from "@vue/test-utils";
import { createTestingPinia } from '@pinia/testing'
import { createRouter, createWebHistory } from "vue-router";
import { flushPromises } from "@vue/test-utils";
import routes from "./routes";
import component from "./App.vue";

const router = createRouter({
  history: createWebHistory(),
  routes,
});

vi.mock("../services/todos", () => ({
  default: function Todos() {
      this.index = vi.fn().mockResolvedValue({data: [{ id: 1 }, { id: 2 }, { id: 3 }]});
    },
}));


describe("App", () => {
  beforeEach(async () => {
    router.push("/");
    await router.isReady();
  });

  const global = {
    plugins: [router, createTestingPinia()],
  };

  it("mounts without error", async () => {
    const wrapper = mount(component, { global });
    await flushPromises()
    expect(wrapper.exists).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

});
