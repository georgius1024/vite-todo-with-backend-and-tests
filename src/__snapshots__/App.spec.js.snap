// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > mounts without error 1`] = `
"<div data-v-7a7a37b1=\\"\\" class=\\"app v-100 h-100\\">
  <div data-v-7a7a37b1=\\"\\" class=\\"panel\\">
    <h1>Todos</h1>
    <div data-v-e895cfd1=\\"\\">
      <div data-v-e895cfd1=\\"\\" class=\\"list-group\\"></div>
      <ul data-v-e895cfd1=\\"\\" class=\\"pagination mt-4\\"></ul>
    </div><a href=\\"/new\\" class=\\"btn btn-secondary\\" role=\\"button\\">New</a>
  </div>
</div>"
`;
